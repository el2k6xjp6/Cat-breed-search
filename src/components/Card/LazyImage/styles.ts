import styled from "styled-components";
import Loading from "../../../assets/Spinner.gif";

// background-image: url("${props => props.error ? props.errorSrc : (props.loading ? Loading : props.actualSrc)}");
export const StyledImage = styled.div<{ loading: boolean; actualSrc: string }>`
  background-image: url("${(props) =>
    props.loading ? Loading : props.actualSrc}");
  background-position: center center;
  background-size: cover;
  width: 100%;
  height: 300px;
  object-fit: cover;
  border-top-left-radius: 24px;
  border-top-right-radius: 24px;
`;
