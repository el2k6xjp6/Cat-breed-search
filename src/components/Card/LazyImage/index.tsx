import React from "react";
import { StyledImage } from "./styles";

interface Props {
  isFetching: boolean;
  actualSrc: string;
}
function LazyImage({ actualSrc, isFetching }: Props) {
  const [isImageLoaded, setImageLoaded] = React.useState(false);
  // const [hasError, setHasError] = React.useState(false);

  React.useEffect(() => {
    const img = new Image();

    img.onload = () => setImageLoaded(true);
    // img.onerror = () => setHasError(true);

    img.src = actualSrc;
  }, [actualSrc]);

  return (
    <StyledImage
      actualSrc={actualSrc}
      loading={!isImageLoaded || isFetching}
      // error={hasError}
      // {...props}
    />
  );
}

export default LazyImage;
