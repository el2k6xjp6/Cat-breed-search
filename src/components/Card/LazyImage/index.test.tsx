import LazyImage from ".";
import "jest-styled-components";
import renderer from "react-test-renderer";

const mockProps = {
  isFetching: true,
  actualSrc: "",
};

describe("LazyImage", () => {
  test("snapshot for LazyImage", () => {
    const tree = renderer.create(<LazyImage {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
