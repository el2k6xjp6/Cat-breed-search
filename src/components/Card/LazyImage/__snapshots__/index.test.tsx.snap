// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LazyImage snapshot for LazyImage 1`] = `
.c0 {
  background-image: url("Spinner.gif");
  background-position: center center;
  background-size: cover;
  width: 100%;
  height: 300px;
  object-fit: cover;
  border-top-left-radius: 24px;
  border-top-right-radius: 24px;
}

<div
  className="c0"
  loading={true}
/>
`;
